clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -1921,10 +1928,26 @@ void CXXNameMangler::mangleLambda(const CXXRecordDecl *Lambda) {
   // if the host-side CXX ABI has different numbering for lambda. In such case,
   // if the mangle context is that device-side one, use the device-side lambda
   // mangling number for this lambda.
+
   unsigned Number = Context.isDeviceMangleContext()
                         ? Lambda->getDeviceLambdaManglingNumber()
                         : Lambda->getLambdaManglingNumber();
-  assert(Number > 0 && "Lambda should be mangled as an unnamed class");
+
+  if (const auto CB = Context.getKernelMangleCallback()) {
+    // If this is involved in kernel mangling and we are in the
+    // __builtin_unique_stable_name parts, this function returns 'true', so we
+    // skip the 'number' apend.
```

Comment:
```suggestion
    // skip the 'number' append.
```

Reply:
I apparently didn't push the updated version yesterday... this comment disappears :)